afkak/consumer.py

```diff
@@ -727,8 +735,19 @@ def _handle_fetch_error(self, failure):
             log.debug(
                 "%r: Exhausted attempts: %d fetching messages from kafka: %r",
                 self, self.request_retry_max_attempts, failure)
-            self._start_d.errback(failure)
-            return
+            if failure.check(OffsetOutOfRangeError):
```

Comment:
You need to move this outside the `if (self.request_retry_max_attempts != 0 and` check at line 725/733, or in the default case it'll never trigger.